src/platform/datapath_raw_xdp.c

```diff
@@ -1570,6 +1553,55 @@ CxPlatXdpTx(
     }
 }
 
+static
+void
+CxPlatXdpFlushTxQueue(
+    _In_ XDP_QUEUE* Queue
+    )
+{
+    //
+    // Move packets from the TX queue (i.e. packets from other threads) to the
+    // worker TX queue (used to actually send packets), but only if we don't
+    // already have anything in the worker TX queue (why?).
```

Comment:
@mtfriesen I believe you wrote the original logic (which we are just moving here), so could you answer the "why?" part? Why do we only move the TxQueue to the WorkerTxQueue if the WorkerTxQueue is currently empty?